### ISSUE TITLE
feat:  eza as a library

### ISSUE THREAD

**Issue by REALERvolker1** [eza-community/eza#729]
A few months ago, I made issue #557 in which I was asking for help using eza as a library. I saw that it had a `lib.rs` and a little bit of documentation, but when I asked for help, I learned that it was not intended to be used as a library. In lieu of some sort of LS_ICONS standard, I would like to use `eza` as a library to pretty-format the contents of a folder without having to depend on the user having the up-to-date binary in their $PATH or something.

Basically what I am asking for here is documentation and public methods we can use to use `eza` in our own projects, similar to how [bat](https://docs.rs/bat/0.24.0/bat/) does it. Since `eza` is MIT-licensed, I do not believe that should be a problem, but idk about that front.

I would make a pr, but I'm still new to rust. I would probably end up doing more harm than good.

**Comment by REALERvolker1** [comment]
One of my little "scripts" that I rewrote in rust, [lsdiff-rs](https://github.com/REALERvolker1/homescripts/tree/main/.config/rustcfg/lsdiff-rs), uses the eza binary as a subprocess, which causes issues moving from Arch to Fedora and back. You can see how I tried to use `eza` as a lib, but I didn't really know how to, so I scrapped it.

**Comment by MartinFillon** [comment]
well as stated in #557 by @cafkafk eza is not made to be used as a lib
but if you want you can work on it yourself, this is clearly not a priority
